rfc/rfc-60/rfc-60.md

```diff
@@ -0,0 +1,225 @@
+
+Here are some of the design considerations:
+
+1. Metadata table is not an optional optimization but a pre-requisite for federated storage to work. 
+Since Hudi 0.11 we have enabled metadata table by default and hence this feature can be enabled by
+the users as long as they are not explicitly turning off metadata table, in which case we should throw an exception.
+
+2. Existing tables cannot switch storage strategy without being re-bootstrapped with the new strategy.
+
+3. The Instant metadata (`HoodieCommitMetadata`,`HoodieCleanMetadata` etc.) will always act as the source of file listing
+for metadata table to be populated.
+
+4. `HoodieCommitMetadata` currently stores `file name` instead of complete `file path`. We will have to modify commit
+metadata to store the complete file path instead of just file name, as the files are now distributed across several random
+prefix paths instead of a derivable table/partition path.
+
+5. If there is an error reading from Metadata table, we will not fall back listing from file system.
+
+6. In case of metadata table getting corrupted or lost, we need to have a solution here to reconstruct metadata table
+from the files which distributed using federated storage. We will likely have to implement a file system listing
+logic, that can get all the partition to files mapping by listing all the prefixes under the `Table Storage Path`.
+Following the folder structure of adding table name/partitions under the prefix will help in getting the listing and
+identifying the table/partition they belong to.
+
+### Query Side Integration
+
+Spark, Hive, [Presto](https://github.com/prestodb/presto/commit/ef1fd25c582631513ccdd097e0a654cda44ec3dc), 
+and [Trino](https://github.com/trinodb/trino/pull/10228) are already integrated to use metadata based listing.
+In general, since these query engines are able to use Hudi's metadata table there should ideally be 
+no changes required in terms of making them work with federated storage. Here are some considerations:
+
+- Spark DataSource and Spark SQL queries have been integrated with metadata based listing via the Hudi's custom implementation
+of Spark's FileIndex interface. However, if Spark DataSource queries are used with globbed paths then the FileIndex path
+does not kick in, and it would rely on Spark's `InMemoryFileIndex` to do the file listing with Hudi's path filter applied. Thus,
+these Spark DataSource queries with globbed paths would not work with federated storage.
+
+- Query engines should be able to determine that federated storage is configured, and rely on metadata to list files. It
+should not be user's responsibility to enable metadata listing from query engines side.
+
+- We need to ensure that partition pruning continues to work for the query engines.
+
+### Future Work
+
+- Currently, we are only targeting Hudi `Data files` to be stored in federated storage format. However, Hudi also
```

Comment:
I remember suggesting timeline-server-based markers, so it's no longer an issue.